File: 0982-minimum-increment-to-make-array-unique/0982-minimum-increment-to-make-array-unique.java
class Solution {

    public int minIncrementForUnique(int[] nums) {

        Arrays.sort(nums);
        
        int l=0;
        int r=l+1;
        int count=0;
       
        while(r<nums.length){
            if(nums[l]==nums[r]) {
                nums[r]=nums[r]+1;
                count++;
            } 
            else if (nums[l]>nums[r]){
                count+=nums[l]-nums[r]+1;
                nums[r]=nums[l]+1;

            }
            r++;
            l++;
        }
        return count;
    }
}

File: 0982-minimum-increment-to-make-array-unique/0982-minimum-increment-to-make-array-unique.py
class Solution:
    def minIncrementForUnique(self, nums: List[int]) -> int:
        nums.sort()
        l,r=0,1
        count=0
        while r<len(nums):
            if nums[l]==nums[r]:
                nums[r]=nums[r]+1
                count+=1
            elif nums[l]>nums[r]:
                count+=nums[l]-nums[r]+1
                nums[r]=nums[l]+1
                
            r+=1
            l+=1
        print(nums)
        return count
            

       